assets/css/base/_base.scss

```diff
@@ -1556,7 +1556,7 @@ button.menu-toggle {
 			font-size: ms(2);
 		}
 
-		a:not( .button ):not( .components-button ) {
+		a:not( .button ):not( .components-button ):not( .wp-block-button__link ) {
 			@include underlinedLink();
```

Comment:
These rules all look backwards to me – can you add a comment about the intention for each?